File: app/src/main/java/com/jonasgerdes/stoppelmap/usecases/transportation/station_detail/DepartureDayFragmentAdapter.java
package com.jonasgerdes.stoppelmap.usecases.transportation.station_detail;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.jonasgerdes.stoppelmap.R;
import com.jonasgerdes.stoppelmap.model.transportation.Station;

/**
 * Created by Jonas on 15.07.2016.
 */
public class DepartureDayFragmentAdapter extends FragmentPagerAdapter {

    private final String[] mPageTitles;
    private final Station mStation;

    public DepartureDayFragmentAdapter(Context context, FragmentManager fm, Station station) {
        super(fm);
        mPageTitles = context.getResources().getStringArray(R.array.days);
        mStation = station;
    }

    @Override
    public Fragment getItem(int position) {
        return DepartureDayFragment.newInstance(mStation, position);
    }

    @Override
    public int getCount() {
        return 6;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mPageTitles[position];
    }
}

File: app/src/main/java/com/jonasgerdes/stoppelmap/usecase/transportation/station_detail/view/DepartureDayFragmentAdapter.kt
package com.jonasgerdes.stoppelmap.usecase.transportation.station_detail.view

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.jonasgerdes.stoppelmap.App
import com.jonasgerdes.stoppelmap.R
import com.jonasgerdes.stoppelmap.model.entity.Station
import com.jonasgerdes.stoppelmap.usecase.transportation.station_detail.view.departure_day.DepartureDayFragment
import com.jonasgerdes.stoppelmap.util.asset.StringResourceHelper
import javax.inject.Inject

/**
 * @author Jonas Gerdes <dev@jonasgerdes.com>
 * @since 10.08.17
 */
class DepartureDayFragmentAdapter(fm: FragmentManager?, val station: Station)
    : FragmentPagerAdapter(fm) {

    @Inject lateinit var stringRes: StringResourceHelper

    init {
        App.graph.inject(this)
    }

    override fun getItem(position: Int): Fragment {
        return DepartureDayFragment.createInstance(station, position)
    }

    override fun getCount(): Int {
        return 6
    }

    override fun getPageTitle(position: Int): CharSequence {
        return stringRes.getArray(R.array.days)[position]
    }
}